### ISSUE TITLE
Class of `gsDesign2` and `simtrial`

### ISSUE THREAD

**Issue by LittleBeannie** [Merck/gsDesign2#457]
In the current version of gsDesign2, we have the class of 

- `fixed_design`
- `gs_design`
- `ahr`
- `wlr`
- `maxcombo`
- `non_binding`
- ...

In the current version of simtrial, we have the class of 

- `counting_process` (to tell whether it is a counting process or TTE datasets)
- `simtrial_gs_wlr` (to summary simulations with >=2 analyses)
- `simtrial_fixed_wlr` (future; to summary simulations with 1 analysis)
- ...

Two questions:

1. Will additional classes be included in gsDesign2 or simtrial?
2. Some keywords, such as 'gs', 'fixed', and 'wlr', are common to both gsDesign2 and simtrial. What is the recommended naming convention to distinguish classes from gsDesign2 or simtrial?

**Comment by jdblischak** [comment]
I agree this is an important discussion to have. The class system in {gsDesign2} is complex.

Some additional high-level questions I have:

1. How are the classes used? In other words, which classes have registered S3 methods, and which are detected inside of functions with `inherits()` or `%in% class()`? And for the latter, can we redesign the functions to use S3 methods instead of these ad-hoc class-specific code chunks?
2. How are the classes related to each other? In other words, are some classes always accompanied by other classes, and thus are more like sub-classes? If yes, then this would inform the potential redesign from the previous point

**Comment by LittleBeannie** [comment]
Temporary suggestions:

simtrial classes
- sim_fixed_ahr
- sim_gs_ahr
- sim_fixed_wlr
- sim_gs_wlr

gsDesign2 classes
- design_fixed_ahr
- design_fixed_wlr
- design_fixed_maxcombo
- design_gs_ahr
- design_gs_wlr
- design_gs_maxcombo
